caom2repo/tests/test-integration.py

```diff
@@ -0,0 +1,127 @@
+#  any later version.                   toute version ultérieure.
+#
+#  OpenCADC is distributed in the       OpenCADC est distribué
+#  hope that it will be useful,         dans l’espoir qu’il vous
+#  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+#  without even the implied             GARANTIE : sans même la garantie
+#  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+#  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+#  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+#  General Public License for           Générale Publique GNU Affero
+#  more details.                        pour plus de détails.
+#
+#  You should have received             Vous devriez avoir reçu une
+#  a copy of the GNU Affero             copie de la Licence Générale
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#  $Revision: 4 $
+#
+# ***********************************************************************
+#
+
+""" Defines TestCaom2IdGenerator class """
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import unittest
+import binascii
+import os
+import sys
+
+from cadcutils.net import auth
+from cadcutils.util import utils
+from caom2repo import CAOM2RepoClient
+from io import BytesIO
+from caom2 import common
+from caom2 import chunk
+from caom2 import part
+from caom2 import plane
+from caom2 import artifact
+from caom2 import observation
+from caom2 import ObservationReader
+from datetime import datetime
+
+THIS_DIR = os.path.dirname(os.path.realpath(__file__))
+
+class TestCaom2Integration(unittest.TestCase):
+            
+    def test_create_and_visit(self):
+        env_a = os.environ['A']
+        cert_file = env_a + '/test-certificates/x509_CADCAuthtest1.pem'
+        subject = auth.Subject(certificate=cert_file)
+        client = CAOM2RepoClient(subject)
+        #client = CAOM2RepoClient(subject, host='rc.cadc-ccda.hia-iha.nrc-cnrc.gc.ca')
+        
+        start = datetime.now()
+        
+        # create one observation for today
+        algorithm = observation.SimpleObservation._ALGORITHM
+        name = obs_name = 'caom2pyinttest{}'.format(start.microsecond)
+        print("test obs name {}".format(name))
+        obs = observation.SimpleObservation("TEST", obs_name)
+        obs.algorithm = algorithm
+        client.put_observation(obs)
+        
+        plugin = os.path.join(THIS_DIR, 'visitor-plugin.py')
+        (visited, updated, skipped, failed) = client.visit(plugin, 'TEST', start=start, halt_on_error=True)
+        print("visit count: {}".format(visited))
+        
+        try:
+            client.delete_observation("TEST", name)
```

Comment:
If we want to keep things clean we should put this in a finally so that observations are always deleted (or deletes are attempted), what do you think?